driver/csiplugin/controllerserver.go

```diff
@@ -1818,3 +1827,108 @@ func (cs *ScaleControllerServer) ControllerExpandVolume(ctx context.Context, req
+			glog.V(5).Infof("cluster details found from cache map for cluster %s are expired.", clusterName)
+			cID := clusterDetails.(ClusterDetails).id
+			conn, err := cs.getConnFromClusterID(cID)
+			if err != nil {
+				return "", err
+			}
+			clusterSummary, err := conn.GetClusterSummary()
+			if err != nil {
+				return "", err
+			}
+			cName := clusterSummary.ClusterName
+			if cName == clusterName {
+				glog.V(5).Infof("updating cluster details in cache map for cluster %s.", clusterName)
+				cs.Driver.clusterMap.Store(ClusterName{cName}, ClusterDetails{cID, cName, time.Now(), 24})
+				cs.Driver.clusterMap.Store(ClusterID{cID}, ClusterDetails{cID, cName, time.Now(), 24})
+				return cID, nil
+			} else {
+				found = false
+			}
+		}
+	}
+
+	if !found {
+		glog.V(5).Infof("cluster details are either expired or not found in cache map for cluster %s. Updating the cache map.", clusterName)
+		scaleconfig := settings.LoadScaleConfigSettings()
+
+		for i := range scaleconfig.Clusters {
+
+			var cID string
+			var cName string
+
+			glog.V(5).Infof("creating new connector for the cluster %s", scaleconfig.Clusters[i].ID)
+			clusterConnector, err := connectors.NewSpectrumRestV2(scaleconfig.Clusters[i])
+			if err != nil {
+				glog.V(5).Infof("unable to create new connector for the cluster %s", scaleconfig.Clusters[i].ID)
+				continue
+			}
+
+			cID = scaleconfig.Clusters[i].ID
+			glog.V(5).Infof("fetching cluster details from cache map for cluster %s", scaleconfig.Clusters[i].ID)
+			clusterDetails, found := cs.Driver.clusterMap.Load(ClusterID{cID})
+			if found {
+				glog.V(5).Infof("checking if cluster details found from cache map for cluster %s has expired.", scaleconfig.Clusters[i].ID)
+				updateTime := clusterDetails.(ClusterDetails).lastupdated
+				expiryDuration := clusterDetails.(ClusterDetails).expiryDuration
+				if time.Since(updateTime).Hours() < float64(expiryDuration) {
+					glog.V(5).Infof("cluster details found from cache map for cluster %s are valid.", scaleconfig.Clusters[i].ID)
+					cName = clusterDetails.(ClusterDetails).name
+					if cName == clusterName {
+						return cID, nil
+					}
+				} else {
+					glog.V(5).Infof("cluster details found from cache map for cluster %s are expired.", scaleconfig.Clusters[i].ID)
+					clusterSummary, err := clusterConnector.GetClusterSummary()
+					if err != nil {
+						glog.V(5).Infof("unable to get cluster summary for cluster %s", scaleconfig.Clusters[i].ID)
+						continue
+					}
+					cName = clusterSummary.ClusterName
+					// cID = fmt.Sprint(clusterSummary.ClusterID)
+					glog.V(5).Infof("updating cluster details in cache map for cluster %s.", scaleconfig.Clusters[i].ID)
+					cs.Driver.clusterMap.Store(ClusterName{cName}, ClusterDetails{cID, cName, time.Now(), 24})
+					cs.Driver.clusterMap.Store(ClusterID{cID}, ClusterDetails{cID, cName, time.Now(), 24})
+					if cName == clusterName {
+						return cID, nil
+					}
+				}
+			}
+
+			if !found {
```

Comment:
This is else part of "if found {" above, why not use else here?

Reply:
using `else` now.